src/core/DY.Core.Bytes.fst

```diff
@@ -1707,6 +1707,46 @@ let pk_dec_enc key nonce msg =
   normalize_term_spec pk_enc;
   normalize_term_spec pk
 
+/// Ghost reverse function for predicates
+
+val pk_enc_reverse_msg: enc_msg:bytes -> GTot (option bytes)
```

Comment:
We may want to think a bit about the names of these functions, to try to emphasize that they aren't to be used in modelling. I'm not sure what a good name to indicate that is, though. Previously we called these something like `invert_pk_enc` (I remember `invert_hash` definitely, but not exactly what the other versions were called), but those also extracted all the components, not just a plaintext. I do broadly like invert better than reverse, but I'm not sure that it's clear enough still.